python/python/res/job_queue/job_manager.py

```diff
@@ -470,21 +482,48 @@ def addLogLine(self, job):
     def runJob(self, job):
         assert_file_executable(job.get('executable'))
         self.addLogLine(job)
+
+        status = job["status"]
+        status.start_time = dt.now()
+        status.status = "Running"
+        self.job_status.dump()
```

Comment:
This feels a bit weird to me, but maybe it's because I don't know these classes sufficiently well. 
If I understood correctly, `runJob` receives a job and executes it. `runJob` belongs to (an instance of) `JobManager`, which has a `self.job_list` member.
As far as I can see here, there is no explicit relation between the input `job` and the member `self.job_list`.
So, when we write `self.job_status.dump()` we are kind of relying on the fact that somehow, the `job` we have in input has been added to the `self.job_status` object.
I'd prefer something like this: 
```Python 
def runJob(self, job):
         ...
         i = self.job_list.index(job)
         status = self.job_status[i]
         ... #do stuff with the status
         self.job_status.dump() # we know for sure that the status we modified is part of job_status
```

Reply:
> This feels a bit weird to me

It is weird - this code is more than ten years old, and in dire need for refactoring and cleanup. The last year we have made changes which should make such improvements easier; it will come - but not in this PR.